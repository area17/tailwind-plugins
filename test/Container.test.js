const generatePluginCss = require('./generatePluginCss');
const cssMatcher = require('jest-matcher-css');
const { Container } = require('../index');

expect.extend({
  toMatchCss: cssMatcher
});

describe('container plugin', () => {
  it('exists', () => {
    expect(typeof Container).toBe('function');
  });

  test('generates correct css', () => {
    return generatePluginCss(Container, {
      theme: {
        screens: {
          xs: { max: '543px' },
          sm: '544px',
          md: '768px',
          lg: '1024px',
          xl: '1440px'
        },
        mainColWidths: {
          xs: 'fluid',
          sm: 'fluid',
          md: 'fluid',
          lg: 'fluid',
          xl: '1220px'
        },
        outerGutters: {
          xs: '16px',
          sm: '16px',
          md: '24px',
          lg: '36px',
          xl: '60px'
        }
      }
    }).then((css) => {
      expect(css).toMatchCss(`
      .container {
        margin-right: auto;
        margin-left: auto;
        width: calc(100vw - (16px * 2));
      }
      @media (min-width: 544px) {
      .container {
          width: calc(100vw - (16px * 2));
      }
      }
      @media (min-width: 768px) {
      .container {
          width: calc(100vw - (24px * 2));
      }
      }
      @media (min-width: 1024px) {
      .container {
          width: calc(100vw - (36px * 2));
      }
      }
      @media (min-width: 1440px) {
      .container {
          width: 1220px;
          padding-right: 0;
          padding-left: 0;
      }
      }
      `);
    });
  });

  test("doesn't output anything when no options set", () => {
    return generatePluginCss(Container, {}).then((css) => {
      expect(css).toMatchCss(``);
    });
  });
});
