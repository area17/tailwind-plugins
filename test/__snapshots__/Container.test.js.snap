// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container plugin generates correct css 1`] = `
".container {
    width: calc(var(--container-width, 100%) - (2 * var(--breakout-container-outer-gutter, var(--container-outer-gutter, var(--outer-gutter, 0)))));
    margin-right: auto;
    margin-left: auto
}
.container > * {
    --container-outer-gutter: 0;
    --breakout-container-outer-gutter: 0
}
.container-reset {
    width: unset;
    margin-right: unset;
    margin-left: unset
}
.container-reset > * {
    --container-outer-gutter: var(--outer-gutter, 0);
    --breakout-container-outer-gutter: inherit
}
.breakout {
    --breakout-outer-gutter: max(var(--outer-gutter), calc((100% - var(--container-width, 100%)) / 2));
    --breakout-container-outer-gutter: var(--outer-gutter);
    position: relative;
    left: 50%;
    width: calc(100vw - var(--scrollbar-visible-width, 0px));
    margin-left: calc((100vw - var(--scrollbar-visible-width, 0px)) / -2)
}
.breakout.px-outer-gutter, .breakout > .px-outer-gutter {
    padding-left: var(--breakout-outer-gutter);
    padding-right: var(--breakout-outer-gutter)
}
.breakout.pr-outer-gutter, .breakout > .pr-outer-gutter {
    padding-right: var(--breakout-outer-gutter)
}
.breakout.pl-outer-gutter, .breakout > .pl-outer-gutter {
    padding-left: var(--breakout-outer-gutter)
}
.breakout > .w-outer-gutter {
    width: var(--breakout-outer-gutter)
}
.breakout-reset {
    --breakout-outer-gutter: var(--outer-gutter);
    --breakout-container-outer-gutter: 0;
    position: unset;
    left: unset;
    width: unset;
    margin-left: unset
}
@media (min-width: 1024px) {
    .lg\\:container {
        width: calc(var(--container-width, 100%) - (2 * var(--breakout-container-outer-gutter, var(--container-outer-gutter, var(--outer-gutter, 0)))));
        margin-right: auto;
        margin-left: auto
    }
    .lg\\:container > * {
        --container-outer-gutter: 0;
        --breakout-container-outer-gutter: 0
    }
    .lg\\:container-reset {
        width: unset;
        margin-right: unset;
        margin-left: unset
    }
    .lg\\:container-reset > * {
        --container-outer-gutter: var(--outer-gutter, 0);
        --breakout-container-outer-gutter: inherit
    }
    .lg\\:breakout {
        --breakout-outer-gutter: max(var(--outer-gutter), calc((100% - var(--container-width, 100%)) / 2));
        --breakout-container-outer-gutter: var(--outer-gutter);
        position: relative;
        left: 50%;
        width: calc(100vw - var(--scrollbar-visible-width, 0px));
        margin-left: calc((100vw - var(--scrollbar-visible-width, 0px)) / -2)
    }
    .lg\\:breakout.px-outer-gutter,.lg\\:breakout > .px-outer-gutter {
        padding-left: var(--breakout-outer-gutter);
        padding-right: var(--breakout-outer-gutter)
    }
    .lg\\:breakout.pr-outer-gutter,.lg\\:breakout > .pr-outer-gutter {
        padding-right: var(--breakout-outer-gutter)
    }
    .lg\\:breakout.pl-outer-gutter,.lg\\:breakout > .pl-outer-gutter {
        padding-left: var(--breakout-outer-gutter)
    }
    .lg\\:breakout > .w-outer-gutter {
        width: var(--breakout-outer-gutter)
    }
    .lg\\:breakout-reset {
        --breakout-outer-gutter: var(--outer-gutter);
        --breakout-container-outer-gutter: 0;
        position: unset;
        left: unset;
        width: unset;
        margin-left: unset
    }
}"
`;
